components/FileInput.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useRef, useState } from "react";
+import Image from "next/image";
+import Plus from "@/assets/images/icons/ic_plus.svg";
+
+interface FileInputProps<T> {
+  name: keyof T;
+  value: string;
+  onChange: (name: keyof T, value: string | null) => void;
+}
+
+const FileInput = <T,>({ name, value, onChange }: FileInputProps<T>) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [preview, setPreview] = useState<string | null>(value);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0] || null;
+    const imageUrl = file ? URL.createObjectURL(file) : null;
+    onChange(name, imageUrl);
+
+    if (file) {
+      setPreview(imageUrl);
+    } else {
+      setPreview(null);
+    }
+  };
```

Comment:
어차피 `file`이 없으면 `imageUrl`을 계산할 수 없으므로,

```suggestion
const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const file = e.target.files?.[0];
    
    if (!file) {
      setPreview(null);
      onChange(name, null);
    } else {
      const imageUrl = URL.createObjectURL(file);
      setPreview(imageUrl);
      onChange(name, imageUrl);
    }
  };
```

처럼 수정해도 좋을 듯 합니다.